aten/src/ATen/native/cuda/ForeachFunctors.cuh

```diff
@@ -287,6 +287,48 @@ struct UnaryOpFunctor {
         }
 };
 
+template<typename T, int depth, int r_args_depth, int res_arg_index>
+struct FracFunctor {
```

Comment:
Why is this different from UnaryOpFunctor?

Reply:
Please, see line 313. Im following the implementation from [here](https://github.com/pytorch/pytorch/blob/9ea7a6c7c55484b83b8cedb58158f429425fe950/aten/src/ATen/native/cuda/UnaryFractionKernels.cu#L35)

Reply:
Its worth pointing out that im planning to refactor this file once again and pass line 313 into some generic block of code.

Reply:
So the op being passed is the Trunc struct:
```
template<typename T>
struct Trunc {
    __device__ T operator()(T t) const { return std::trunc(t); }
};
```
Why not create a struct that computes frac (via t - std::trunc(t)) and pass that into UnaryOpFunctor?

Reply:
@zou3519, you are right. rewrote to use UnaryOpFunctor. FracFunctor is gone